Always pass CFLAGS (which is really CXXFLAGS) to the C++ compiler.
This is necessary to get the -stdlib flag.
--- ann2fig/Makefile.orig	2010-01-27 22:40:01.000000000 -0600
+++ ann2fig/Makefile	2020-01-05 21:11:27.000000000 -0600
@@ -61,7 +61,7 @@
 targets: $(BINDIR)/$(ANN2FIG)

 $(BINDIR)/$(ANN2FIG): $(OBJECTS)
-	$(C++) $(OBJECTS) -o $(ANN2FIG) $(LDFLAGS) $(ANNLIBS) $(OTHERLIBS)
+	$(C++) $(CFLAGS) $(OBJECTS) -o $(ANN2FIG) $(LDFLAGS) $(ANNLIBS) $(OTHERLIBS)
 	mv $(ANN2FIG) $(BINDIR)

 #-----------------------------------------------------------------------------
@@ -75,7 +75,7 @@
 #-----------------------------------------------------------------------------
 
 ann2fig.o: ann2fig.cpp
-	$(C++) -c -I$(INCDIR) ann2fig.cpp
+	$(C++) -c -I$(INCDIR) $(CFLAGS) ann2fig.cpp
 
 #-----------------------------------------------------------------------------
 # Cleaning
--- sample/Makefile.orig	2010-01-27 22:40:01.000000000 -0600
+++ sample/Makefile	2020-01-05 21:10:58.000000000 -0600
@@ -64,7 +64,7 @@
 targets: $(BINDIR)/$(ANNSAMP)

 $(BINDIR)/$(ANNSAMP): $(SAMPOBJECTS) $(LIBDIR)/$(ANNLIB)
-	$(C++) $(SAMPOBJECTS) -o $(ANNSAMP) $(LDFLAGS) $(ANNLIBS)
+	$(C++) $(CFLAGS) $(SAMPOBJECTS) -o $(ANNSAMP) $(LDFLAGS) $(ANNLIBS)
 	mv $(ANNSAMP) $(BINDIR)

 #-----------------------------------------------------------------------------
--- test/Makefile.orig	2010-01-27 22:40:01.000000000 -0600
+++ test/Makefile	2020-01-05 21:07:26.000000000 -0600
@@ -67,7 +67,7 @@
 targets: $(BINDIR)/$(ANNTEST)
 
 $(BINDIR)/$(ANNTEST): $(TESTOBJECTS) $(LIBDIR)/$(ANNLIB)
-	$(C++) $(TESTOBJECTS) -o $(ANNTEST) $(LDFLAGS) $(ANNLIBS) $(OTHERLIBS)
+	$(C++) $(CFLAGS) $(TESTOBJECTS) -o $(ANNTEST) $(LDFLAGS) $(ANNLIBS) $(OTHERLIBS)
 	mv $(ANNTEST) $(BINDIR)
 
 #-----------------------------------------------------------------------------
